web_infrastructure/jboss_cli.py

```diff
@@ -0,0 +1,137 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+DOCUMENTATION = '''
+---
+module: jboss_cli
+version_added: "2.1"
+author: "Roland Wolters (@liquidat)"
+short_description: Manage JBoss via jboss_cli
+requirements:
+    - jboss_cli
+description:
+    - This module provides JBoss management via jboss_cli for JBoss application servers
+options:
+    user:
+        description:
+            - the name of the management user
+        required: false
+    password:
+        description:
+            - the password to assign to the username
+        required: false
+    host:
+        description:
+            - the host running JBoss
+        required: false
+        default: '127.0.0.1'
+    port:
+        description:
+            - the port of the associated host running JBoss
+        required: false
+        default: '9999'
+    jbosshome:
+        description:
+            - The JBoss path environment variable
+        required: false
+        default: '/usr/share/jbossas/'
+    command:
+        description:
+            - The command to manage the JBoss server
+        required: true
+'''
+EXAMPLES = '''
+Example playbook entries using the jboss_cli module to manage JBoss.
+
+    tasks:
+
+    - name: set max-threads to a given value
+      jboss_cli: command="/subsystem=ejb3/thread-pool=default:write-attribute(name=max-threads,value=3000)"
+
+    - name: get all configuration and runtime details
+      jboss_cli:
+        user: jbossuser
+        password: somepassword
+        host: 127.0.0.1
+        port: 1234
+        command: ":read-resource(include-runtime=true, recursive=true, recursive-depth=10)"
+        jbosshome: "/usr/share/jbossas/"
+
+    - name: deploy a given war
+      jboss_cli:
+        user: jbossuser
+        password: jbosspassword
+        host: 10.10.10.10
+        port: 1234
+        command: "deploy /path-up-to/your-app.war"
+'''
+RETURN = '''
+msg:
+    description: execution results from the jboss-cli
+    returned: success, failure
+    type: string
+    sample: outcome => success
+'''
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            command   = dict(required=True, type='str'),
+            user      = dict(required=False, type='str', default=None),
+            password  = dict(required=False, type='str', default=None, no_log=True),
+            host      = dict(required=False, type='str', default='127.0.0.1'),
+            port      = dict(required=False, type='str', default='9999'),
+            jbosshome = dict(required=False, type='str', default='/usr/share/jbossas/'),
+        ),
+        # we are not there yet
+        #supports_check_mode = True,
```

Comment:
Could you please add some details (in the code) here explaining why check_mode isn't available.
The comment as it stands reads like it _may_ be possible to add check_mode in the future, where as the PR description implies it isn't possible.